src/pages/AssetPage/index.tsx

```diff
@@ -0,0 +1,277 @@
+          <div className="transactions-table">
+            {this.showEmptyState && (
+              <div className="empty-state">
+                You don't have any transactions yet
+              </div>
+            )}
+            <InfiniteScroll
+              loadMore={this.handleLoadMoreTransactions}
+              hasMore={this.hasMoreTransactions}
+            >
+              {this.showTransactionsTable && (
+                <Table responsive>
+                  <thead>
+                    <tr>
+                      <th>Asset</th>
+                      <th>Date</th>
+                      <th>Operation</th>
+                      <th>Status</th>
+                      <th>Amount</th>
+                    </tr>
+                  </thead>
+                  <tbody>
+                    {this.transactionStore.assetTransactions.map(t => (
+                      <tr key={t.id}>
+                        <td>
+                          <div className="asset-col">
+                            <img
+                              width="48"
+                              src={asset.iconUrl || ASSET_DEFAULT_ICON_URL}
+                            />
+                            <div>
+                              <div className="asset-col__asset_name">
+                                {asset.name}
+                              </div>
+                              <div className="asset-col__wallet_name">
+                                Trading Wallet
+                              </div>
+                            </div>
+                          </div>
+                        </td>
+                        <td>
+                          <FormattedDate
+                            day="2-digit"
+                            month="2-digit"
+                            year="2-digit"
+                            value={t.dateTime}
+                          />, <FormattedTime value={t.dateTime} />
+                        </td>
+                        <td>{TransactionTypeLabel[t.type]}</td>
+                        <td>{TransactionStatusLabel[t.state]}</td>
+                        <td>
+                          <ColoredAmount
+                            value={t.amount}
+                            accuracy={asset.accuracy}
+                            assetName={asset.name}
+                          />
+                        </td>
+                      </tr>
+                    ))}
+                  </tbody>
+                </Table>
+              )}
+            </InfiniteScroll>
+            {this.areTransactionsLoading && <Spinner />}
+            {this.showLoadMoreButton && (
+              <div
+                className="show-more-button"
+                onClick={this.handleLoadMoreTransactions}
+              >
+                Show more...
+              </div>
+            )}
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  private loadTransactions = async () => {
```

Comment:
why not move it into the corresponding `(transaction)Store`?

Reply:
Fetching is performing in transactionStore (fetchAssetTransactions method), but there are few page specific things like get asset from URL, show loader, which, i believe, should be on the page level